### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by jimohafeezco**
unit and integration test implemnetation

### CODE REVIEW

File: scripts/inTest.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        self.assertNotAlmostEqual(self.data1,0)
+        self.assertNotAlmostEqual(self.data2,0)
+        self.assertNotAlmostEqual(self.data3,0)
+
+        # rospy.loginfo("I heard %s :", self.data)
+ 
+def listener():
+    rospy.init_node('listener', anonymous=True)
+
+    rospy.Subscriber("spot/imu_data", Imu, MyTestCase().callback)
```
Comment (l1va) [inline scripts/inTest.py:32]: integration test is not that i asked.

File: scripts/FKtest.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        np.testing.assert_allclose(test1, FK1, rtol = 0.0001, atol =0)
+        np.testing.assert_allclose(test2, FK2, rtol = 0.0001, atol =0)
+    def test2(self):
+        np.testing.assert_allclose(test3, FK3, rtol = 0.0001, atol =0)
+        np.testing.assert_allclose(test4, FK3, rtol = 0.0001, atol =0)
+    def test3(self):
+        self.assertNotEqual(test1.any(), 0)
+        self.assertNotEqual(test2.any(), 0)
+        self.assertNotEqual(test3.any(), 0)
+        self.assertNotEqual(test4.any(), 0)
```
Comment (l1va) [inline scripts/FKtest.py:65]: interesting structure, might be useless testing, but i see asserts and everything that i want for unit test. That is why will discuss why it seems not so good, but practically all is ok.

File: scripts/FKine.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import numpy as np
+#from tf import transformations
+#
+
+def DHMatrix(alpha,a,d,theta):
+    return np.matrix([[cos(theta), -sin(theta), 0, a],
+                   [sin(theta)*cos(alpha), cos(theta)*cos(alpha), -sin(alpha), -sin(alpha)*d],
+                   [sin(theta)*sin(alpha), cos(theta)*sin(alpha), cos(alpha), cos(alpha)*d],
+                   [0, 0, 0, 1]])
+
```
Comment (l1va) [inline scripts/FKine.py:23]: even DH, great =)

File: README.md [hunk 4]
```diff
@@ -57,3 +57,27 @@ To get Camera Images,run:
...
+
+```python FKtest.py```
+
+
+### Integration Test
+
+
+The integration tests is a continuation of the joint movement/control task.
+Firstly, from previous task, some joints poitions are published to robot joints. In other to get the actual joint position, an IMU sensor was added to the urdf file as a Gazebo plugin. Thereafter a subscriber was implemneted to determine if actual movement takes place which is determined by reading accelearation information from the robot joint. The integration test confirms if actual movement takes place when some joint positions are published. 
+
```
Comment (l1va) [inline README.md:78]: hm, now i see, but better to read joint_states, because sensor can work and send values but the positions is still zero.

**Review by l1va — CHANGES_REQUESTED** [review]

File: scripts/FKine.py (hunk 3 above)
Comment (jimohafeezco) [reply scripts/FKine.py:23]: okay

File: README.md (hunk 4 above)
Comment (jimohafeezco) [reply README.md:78]: yes, position I still zero. I actually didn't figure out how to read exact joint positions. So i decided to measure linear acceleration. So the test is basically checking if linear acceleration is no longer zero once joint commands have been sent.

**Pull request merged**